java-squid/src/main/java/org/sonar/java/se/checks/NullDereferenceCheck.java

```diff
@@ -65,19 +65,30 @@ public ProgramState checkPreStatement(CheckerContext context, Tree syntaxNode) {
       return context.getState();
     }
     if (syntaxNode.is(Tree.Kind.MEMBER_SELECT)) {
-      if ("class".equals(((MemberSelectExpressionTree) syntaxNode).identifier().name())) {
-        // expression ClassName.class won't raise NPE.
-        return context.getState();
+      return checkMemberSelect(context, (MemberSelectExpressionTree) syntaxNode, currentVal);
+    }
+    if (syntaxNode.is(Tree.Kind.METHOD_INVOCATION)) {
+      MethodInvocationTree methodInvocation = (MethodInvocationTree) syntaxNode;
+      if (methodInvocation.methodSelect().is(Tree.Kind.MEMBER_SELECT)) {
+        return checkMemberSelect(context, (MemberSelectExpressionTree) methodInvocation.methodSelect(), currentVal);
       }
+    }
+    return context.getState();
+  }
+
+  private ProgramState checkMemberSelect(CheckerContext context, MemberSelectExpressionTree syntaxNode, SymbolicValue currentVal) {
+    if ("class".equals(syntaxNode.identifier().name())) {
+      // expression ClassName.class won't raise NPE.
+      return context.getState();
+    }
 
-      if (context.isNull(currentVal)) {
+    if (context.isNull(currentVal)) {
         context.reportIssue(syntaxNode, this, "NullPointerException might be thrown as '" + SyntaxTreeNameFinder.getName(syntaxNode) + "' is nullable here");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 7. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AIndentationCheck)